### PR TITLE
Added support of failed exchanges.

### LINKED ISSUE #3: Add Support for failed messages

### PR DESCRIPTION
**Pull request by tdanylchuk**
fixes https://github.com/Playtika/sleuth-camel/issues/3

### CODE REVIEW

File: sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
     @Override
     public void notify(EventObject event) {
-        log.trace("Caught event [{}] - processing...", event);
+        log.trace("Caught an event [{} - {}] - processing...", event.getClass().getSimpleName(), event);
```
Comment (Aloren) [inline sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java:56]: Caught?

Comment (tdanylchuk) [reply sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java:56]: https://www.dictionary.com/browse/caught

**Comment by tdanylchuk** [comment]
image

File: sleuth-camel-core/src/test/java/com/playtika/sleuth/camel/functional/FunctionalTest.java [hunk 2]
```diff
@@ -168,14 +182,65 @@ public void shouldSendToRouteWithSpanInHeaders() throws Exception {
...
+        String errorMessage = "Something went wrong";
+
+        directRouteMockEndpoint.expectedMessageCount(0);
+        exceptionRouteMockEndpoint.expectedBodiesReceived(1);
+        exceptionRouteMockEndpoint.expectedBodiesReceived(TEST_BODY);
+
+        doThrow(new RuntimeException(errorMessage)).when(mockProcessor).process(any(Exchange.class));
+
+        try {
+            directRouteProducer.sendBody(TEST_BODY);
```
Comment (jnopnop) [inline sleuth-camel-core/src/test/java/com/playtika/sleuth/camel/functional/FunctionalTest.java]: `assertThatThrownBy(() -> directRouteProducer.sendBody(TEST_BODY))
        .isInstanceOf(CamelExecutionException.class)`

Comment (tdanylchuk) [reply sleuth-camel-core/src/test/java/com/playtika/sleuth/camel/functional/FunctionalTest.java]: thx, done

**Pull request merged**